framework/main.py

```diff
@@ -10,77 +10,122 @@
 import sys
 
 
-def init_argparser():
+def init_argparser(args=None):
     """
     Initialization  of argument parse library with it's arguments
     Args:
-        None
+        args: list of args to parse in `sys.argv` form. Useful for testing
+            passed to `ArgumentParser.parse_args` . default to None (use `sys.argv`)
 
     Returns:
         object: Parsed arguments - returned from parser.parse_args()
     """
+    server_args_parser = ArgumentParser(add_help=False)
+
+    # sharing arguments using parent parsers.
+    # https://stackoverflow.com/a/56595689 and https://stackoverflow.com/a/7498853
 
-    parser = ArgumentParser()
-    parser.add_argument(
+    # TODO These really should be config file things,
+    server_args_parser.add_argument(
+        "-s",
```

Comment:
I am a bit concerned about having one letter shortcut for possibly two reasons (maybe su could be a better option):

1 ) Maybe we might want to save the `-s` for shortening the `signoff` which is already a mandatory argument

2 ) Maybe some other arguments in the future that will be single-worded and start with an `s` as well

Reply:
Would `-u`  for url be better?  

Also I think it might be okay to not shorten it, but i do think we should add support for some sort of config file so you dont need to enter it (and the token, etc) all the time. then we could just have it be `--server`

I was curious does your team commonly deal with multiple resource locker instances in production?